src/libraries/System.Console/src/System/Console.cs

```diff
@@ -960,18 +968,19 @@ public static void Write(string? value)
             Out.Write(value);
         }
 
-        internal static bool HandleBreakEvent(ConsoleSpecialKey controlKey, bool cancel = false)
+        private static void HandlePosixSignal(PosixSignalContext ctx)
         {
-            ConsoleCancelEventHandler? handler = s_cancelCallbacks;
-            if (handler == null)
+            Debug.Assert(ctx.Signal == PosixSignal.SIGINT || ctx.Signal == PosixSignal.SIGQUIT);
+
+            bool cancel = false;
+            if (s_cancelCallbacks is ConsoleCancelEventHandler handler)
             {
-                return false;
+                var args = new ConsoleCancelEventArgs(ctx.Signal == PosixSignal.SIGINT ? ConsoleSpecialKey.ControlC : ConsoleSpecialKey.ControlBreak);
+                handler(null, args);
```

Comment:
Do we want to set `args.Cancel` to match `ctx.Cancel` before invoking the handler?

Reply:
Hmm.  I'd decided no, but, I guess it probably makes sense.  Will fix.